ml/src/main/scala/org/apache/spark/ml/h2o/algos/H2OAutoML.scala

```diff
@@ -135,43 +134,9 @@ class H2OAutoML(override val uid: String) extends Estimator[H2OMOJOModel]
   }
 
   override def copy(extra: ParamMap): this.type = defaultCopy(extra)
-
-  def defaultFileName: String = H2OAutoML.defaultFileName
-}
-
-
-object H2OAutoML extends MLReadable[H2OAutoML] {
-
-  private final val defaultFileName = "automl_params"
-
-  @Since("1.6.0")
-  override def read: MLReader[H2OAutoML] = H2OAutoMLReader.create[H2OAutoML](defaultFileName)
-
-  @Since("1.6.0")
-  override def load(path: String): H2OAutoML = super.load(path)
-}
-
-private[algos] class H2OAutoMLReader[A <: H2OAutoML : ClassTag](val defaultFileName: String) extends MLReader[A] {
-
-  override def load(path: String): A = {
-    val metadata = DefaultParamsReader.loadMetadata(path, sc)
-
-    val algo = make[A](metadata.uid)
-    metadata.getAndSetParams(algo)
-    algo
-  }
-
-  private def make[CT: ClassTag](uid: String): CT = {
-    val aClass = implicitly[ClassTag[CT]].runtimeClass
-    val ctor = aClass.getConstructor(classOf[String])
-    ctor.newInstance(uid).asInstanceOf[CT]
-  }
 }
 
-
-object H2OAutoMLReader {
-  def create[A <: H2OAutoML : ClassTag](defaultFileName: String) = new H2OAutoMLReader[A](defaultFileName)
-}
+object H2OAutoML extends DefaultParamsReadable[py_sparkling.ml.algos.H2OAutoML]
```

Comment:
Can you help me understand why types from ```py_sparkling.ml.algos._``` are passed as generic parameters to ```DefaultParamsReadable```? I would expect types that implement ```DefaultParamsReadable```.

Reply:
This is needed as we always need to store the classes using `py_sparkling.ml.algos` classes.

Without this, it works fine if you just train and run pipelines in PySpark or just in Scala.

But if you, for example, decide to train pipeline in Scala, export it and load it in PySpark then we hid the issue as `org.apache.spark` prefix is translated to `pyspark` on Python side. So to support all cases, we need to use `py_sparkling` classes as backend.

Hope it makes sense